src/cpp/src/icontinuous_batching.hpp

```diff
@@ -70,20 +70,20 @@ class ContinuousBatchingPipeline::IContinuousBatchingPipeline {
     /**
      * Performs a single inference step of all running (and pulls awaiting) requests
      */
-    virtual void step() = 0;
+    virtual size_t step() = 0;
 
     /**
      * Performs monolitic generation based on encoded prompts
      */
-    virtual std::vector<EncodedGenerationResult>
+    virtual std::pair<std::vector<EncodedGenerationResult>, PerfMetrics>
     generate(const std::vector<ov::Tensor>& input_ids,
              const std::vector<GenerationConfig>& sampling_params,
              const StreamerVariant& streamer) = 0;
 
     /**
      * Performs monolitic generation based on text prompts
      */
-    std::vector<GenerationResult>
+    std::pair<std::vector<GenerationResult>, PerfMetrics>
```

Comment:
can we make `PerfMetrics` to be a member of `GenerationResult` / `EncodedGenerationResult` structures? to avoid this ugly `std::pair`

It will be aligned with LLMPipeline structures

Reply:
At first i wanted to do so, but this method returns vector of GenerationResult, but perf metrics are for the whole batch of generated results. We cannot distribute them to all vectors

Reply:
Done. Now perf_metrics are stored in all GenerationResult's with individual tok/detokenization times